spec/cloud_test.go

```diff
@@ -54,8 +57,59 @@ func (g *mockGCECloudMetaGenerator) IsGCE(ctx context.Context) bool {
 }
 
 func TestCloudGenerate(t *testing.T) {
+	generator := &mockCloudMetaGenerator{
+		metadata: &mackerel.Cloud{
+			Provider: "mock",
+			MetaData: map[string]string{
+				"mockKey": "mockValue",
+			},
+		},
+		customIdentifier: "mock-generated-identifier.example.com",
+	}
+	g := &CloudGenerator{generator}
+
+	value, err := g.Generate()
+	if err != nil {
+		t.Errorf("should not raise error: %s", err)
+	}
+
+	cloud, typeOk := value.(*mackerel.Cloud)
+	if !typeOk {
+		t.Errorf("value should be *mackerel.Cloud. %+v", value)
+	}
+
+	if cloud.Provider != "mock" {
```

Comment:
[nit] this expression will panic if previous test was error. but i think ok because it may not be happen.

Reply:
Should we use `t.Fatalf` instead?

Reply:
IMHO, no. `t.Fatalf` can use only when the prerequisites are failed.

I suggest to use `return` statement or `if cloud != nil && cloud.Provider...` if you fix politely.

Reply:
Okay. I'll fix TestCloudGenerate and TestEC2Generator (, but keep others as is).